src/analyses/call_graph_helpers.cpp

```diff
@@ -74,10 +74,10 @@ std::set<irep_idt> get_reaching_functions(
 std::set<irep_idt> get_functions_reachable_within_n_steps(
   const call_grapht::directed_grapht &graph,
   const std::set<irep_idt> &start_functions,
-  std::size_t &n)
+  std::size_t n)
 {
-  std::vector<std::size_t> start_indices;
   std::set<irep_idt> result;
+  std::vector<std::size_t> start_indices;
```

Comment:
This looks a little like an inconsistency when updating.  Where the two PRs that this depended on changed and this undoes some of the changes?

Reply:
One of the PRs that this depends on ended up in a mess, because I did a major requested change, then did some minor changes on top of that, and then had to reverse the major change. And so there are some bits and pieces that have ended up in this PR that shouldn't be, like passing std::size_t by reference. Those changes are deliberate. 

Switching the order of start_indices and result is unintentional though, I've reversed that now

Reply:
Thanks.  I appreciate that some of this is a bit pedantic and my thanks for your patience.  I just want to avoid getting into a fix / undo because PR raced with fix / fix loop.

Reply:
I think @martin-cs' comment should be addressed.